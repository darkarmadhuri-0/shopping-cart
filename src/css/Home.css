#carouselExampleIndicators {
  background-color: red;
  height: 300px;
}
