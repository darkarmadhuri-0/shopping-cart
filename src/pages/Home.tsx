import React from "react";
import "bootstrap/dist/css/bootstrap.css";
import Carousel from "react-bootstrap/Carousel";

export function Home() {
  return (
    <div className="flex justify-center h-[82vh] w-screen border border-black bg-slate-700">
    <div className="block w-[600px] h-[20vh] border border-black bg-red-500">
      {/* style={{ display: "block", width: 700, padding: 30 }} */}
      <h4 className="text-center"><i>OMG!</i><br/><i>DEALS</i><br/>JAW-DROPPING price</h4>
      <Carousel>
        <Carousel.Item interval={1500}>
          <img
            className="w-full h-[400px] object-cover"
            src="/imgs/dress.png"
            alt="Image One"
          />
          <Carousel.Caption>
            <h3 className="text-red-500 font-bold  font-mono italic">FUN,FASHION,SAVING AND MORE</h3>
            <p className="text-red-500 font-bold italic">The DRESS Store<br/>+Explore</p>
          </Carousel.Caption>
        </Carousel.Item>
        <Carousel.Item interval={500}>
          <img
            className="w-full h-[400px]"
            src="/imgs/men.jpg"
            alt="Image Two"
          />
          <Carousel.Caption>
          <h3 className="text-red-500 font-bold font-mono italic ">FUN,FASHION,SAVING AND MORE </h3>
            <p className="text-red-500 font-bold italic">The DRESS Store<br/>+Explore</p>
          </Carousel.Caption>
        </Carousel.Item>
      </Carousel>
    </div>
    </div>
  );
}
