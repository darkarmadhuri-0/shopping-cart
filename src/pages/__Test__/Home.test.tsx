import React from "react";

import { render, screen, cleanup } from "@testing-library/react";

import { RecoilRoot } from "recoil";

import renderer from "react-test-renderer";
import { Home } from "../Home";



describe("Home Page Component testing", () => {

   const HomeTest: React.FC = () => {
    
    return (

      <RecoilRoot>

        <Home />

      </RecoilRoot>

    );

  };

  afterEach(() => {

    cleanup();

  });
  

  it("should dispaly the different deals", () => {

    render(<HomeTest />);

    expect(screen.getByText(/DEALS/i)).toBeInTheDocument();

  });


   it("should match with snapshot of welcome module", () => {

    const tree = renderer.create(<HomeTest />).toJSON();

    expect(tree).toMatchSnapshot();

  });

});