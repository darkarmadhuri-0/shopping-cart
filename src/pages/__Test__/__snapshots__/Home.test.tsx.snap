// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Component testing should match with snapshot of welcome module 1`] = `
<div
  className="flex justify-center h-[82vh] w-screen border border-black bg-slate-700"
>
  <div
    className="block w-[600px] h-[20vh] border border-black bg-red-500"
  >
    <h4
      className="text-center"
    >
      <i>
        OMG!
      </i>
      <br />
      <i>
        DEALS
      </i>
      <br />
      JAW-DROPPING price
    </h4>
    <div
      className="carousel slide"
      onKeyDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="carousel-indicators"
      >
        <button
          aria-current={true}
          aria-label="Slide 1"
          className="active"
          data-bs-target=""
          onClick={[Function]}
          type="button"
        />
        <button
          aria-current={false}
          aria-label="Slide 2"
          data-bs-target=""
          onClick={[Function]}
          type="button"
        />
      </div>
      <div
        className="carousel-inner"
      >
        <div
          className="active carousel-item"
          interval={1500}
        >
          <img
            alt="Image One"
            className="w-full h-[400px] object-cover"
            src="/imgs/dress.png"
          />
          <div
            className="carousel-caption"
          >
            <h3
              className="text-red-500 font-bold  font-mono italic"
            >
              FUN,FASHION,SAVING AND MORE
            </h3>
            <p
              className="text-red-500 font-bold italic"
            >
              The DRESS Store
              <br />
              +Explore
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
          interval={500}
        >
          <img
            alt="Image Two"
            className="w-full h-[400px]"
            src="/imgs/men.jpg"
          />
          <div
            className="carousel-caption"
          >
            <h3
              className="text-red-500 font-bold font-mono italic "
            >
              FUN,FASHION,SAVING AND MORE 
            </h3>
            <p
              className="text-red-500 font-bold italic"
            >
              The DRESS Store
              <br />
              +Explore
            </p>
          </div>
        </div>
      </div>
      <a
        className="carousel-control-prev"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
        <span
          className="visually-hidden"
        >
          Previous
        </span>
      </a>
      <a
        className="carousel-control-next"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
        <span
          className="visually-hidden"
        >
          Next
        </span>
      </a>
    </div>
  </div>
</div>
`;
